datafusion/functions/src/string/concat.rs

```diff
@@ -90,23 +191,124 @@ impl ScalarUDFImpl for ConcatFunc {
 
     fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
         use DataType::*;
-        let mut dt = &Utf8;
-        arg_types.iter().for_each(|data_type| {
-            if data_type == &Utf8View {
-                dt = data_type;
+        
+        // Check if any argument is an array type
+        let has_array = arg_types.iter().any(|dt| {
+            matches!(dt, List(_) | LargeList(_) | FixedSizeList(_, _))
+        });
```

Comment:
Is this mirroring whats done in `coerce_types`? We shouldn't duplicate the logic as the argument inputs to `return_type` are already coerced